plugins/pulsar-records-storage/src/test/java/com/github/bsideup/liiklus/pulsar/PulsarRecordsStorageTest.java

```diff
@@ -70,4 +91,212 @@ public String keyByPartition(int partition) {
     public int getNumberOfPartitions() {
         return NUM_OF_PARTITIONS;
     }
+
+    @Test
+    @Override
+    // since pulsar behave differently for closed ledger, need to overwrite this test to include the 1 entry setback
+    public void testInitialOffsets() throws Exception {
+        var offsetInfos = publishMany("key".getBytes(), 10);
+        var offsetInfo = offsetInfos.get(7);
+        var partition = offsetInfo.getPartition();
+        var position = offsetInfo.getOffset();
+
+        var receivedRecords = subscribeToPartition(partition, Optional.of("earliest"), () -> CompletableFuture.completedFuture(Collections.singletonMap(partition, position)))
+                .flatMap(RecordsStorage.PartitionSource::getPublisher)
+                .take(4)
+                .collectList()
+                .block(Duration.ofSeconds(10));
+
+        assertThat(receivedRecords).extracting(RecordsStorage.Record::getOffset).containsExactly(
+                offsetInfos.get(6).getOffset(),
+                offsetInfos.get(7).getOffset(),
+                offsetInfos.get(8).getOffset(),
+                offsetInfos.get(9).getOffset()
+        );
+    }
+
+    @Test
+    @Override
+    // currently each liiklus consumers consume all partition, and leaves the client to rebalance
+    // in pulsar, the pulsar client does not rebalance naturally on failover or exclusive
+    public void testExclusiveRecordDistribution() throws Exception {
```

Comment:
Same comment as in `testMultipleGroups`

Reply:
same comment as well in `testMultipleGroups`

Reply:
this test verifies that:
1. Eventually, both subscriptions will receive assignments
2. their received records are not overlapping

this is not Kafka specific, and if this test fails - there seems to be a bug in the storage implementation